### PR TITLE
Initial bout of name fixes

### PR DESCRIPTION
**Pull request by luke-hill**
Initial first attempt pass at fixing up https://github.com/natritmeyer/site_prism/issues/315
Still a bit of work to do.

### CODE REVIEW

File: test_site/pages/home.rb [hunk 1]
```diff
@@ -8,59 +8,44 @@ class Home < SitePrism::Page
...
   element :welcome_message, 'body > span'
   element :go_button, '[value="Go!"]'
-  element :link_to_search_page, :xpath, '//p[2]/a'
-  # Below element takes just over 2 seconds to appear
-  element :some_slow_element, :xpath, '//a[@class="slow"]'
-  element :invisible_element, 'input.invisible'
-  element :shy_element, 'input#will_become_visible'
-  element :retiring_element, 'input#will_become_invisible'
+  element :search_page_link, :xpath, '//p[2]/a'
+  element :slow_element, :xpath, '//a[@class="slow"]'
```
Comment (tgaff) [inline test_site/pages/home.rb]: The comment might actually be useful to keep.

File: test_site/pages/home.rb [hunk 2]
```diff
@@ -8,59 +8,44 @@ class Home < SitePrism::Page
...
-  # elements groups
-  elements :removing_links, '#link_container_will_become_nonexistent > a'
-  elements :nonexistent_elements, 'input#nonexistent'
-  elements :welcome_headers, :xpath, '//h3'
-  elements :welcome_messages, :xpath, '//span'
+  # element groups
+  elements :removing_links, '#removing_link_div > a'
+  elements :nonexistents, 'input#nonexistent'
   elements :rows, 'td'
   # note - actually only one of the below
```
Comment (tgaff) [inline test_site/pages/home.rb]: this seems like maybe a bug.... should there be more than one?

File: test_site/pages/home.rb (hunk 1 above)
Comment (luke-hill) [reply test_site/pages/home.rb]: This comment was incorrect, so removed.

**Comment by luke-hill** [comment]
@tgaff @ineverov My html / js isn't up to scratch, so I googled some tips.

I "think" I've written the js functions in a slightly more efficient way, added a 2nd link (Which we never had enumerable slow links before), and amended some timings to make the suite run even quicker.

Let me know if you think this is ok. I've cut 3.0.beta this morning so this will just hit master and the full 3.0 release.

**Pull request merged**